model/system_metrics.go

```diff
@@ -32,6 +32,32 @@ var (
 	systemMetricsArtifactKey    = bsonutil.MustHaveTag(SystemMetrics{}, "Artifact")
 )
 
+// CreateSystemMetrics is the entry point for creating the metadata for
+// system metric time series data for a task execution.
+func CreateSystemMetrics(info SystemMetricsInfo, artifact SystemMetricsArtifactInfo) *SystemMetrics {
+	return &SystemMetrics{
+		ID:        info.ID(),
+		Info:      info,
+		CreatedAt: time.Now(),
+		Artifact: SystemMetricsArtifactInfo{
```

Comment:
Is there any reason we don't just use the artifact info input directly?

Reply:
It needs to set the prefix and key - they shouldn't be specified by user. Unless you mean reuse the object and just set those fields on it?

Reply:
Yeah, normally you would use the user's input directly unless you have some reason to transform the input. In this case, if the prefix/key can't be specified by the user, can you just mention in the doc comment that any prefix/key they set will be overridden?

Otherwise this seems fine.